<Type Name="HttpException" FullName="System.Web.HttpException">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="58c641a30652a7ae02454aaab9977f13384e538a" />
    <Meta Name="ms.sourcegitcommit" Value="434f60616a9793fa8436744549fc856e94f7a648" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="es-ES" />
    <Meta Name="ms.lasthandoff" Value="08/25/2018" />
    <Meta Name="ms.locfileid" Value="39856858" />
  </Metadata>
  <TypeSignature Language="C#" Value="public class HttpException : System.Runtime.InteropServices.ExternalException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit HttpException extends System.Runtime.InteropServices.ExternalException" />
  <TypeSignature Language="DocId" Value="T:System.Web.HttpException" />
  <TypeSignature Language="VB.NET" Value="Public Class HttpException&#xA;Inherits ExternalException" />
  <TypeSignature Language="C++ CLI" Value="public ref class HttpException : System::Runtime::InteropServices::ExternalException" />
  <TypeSignature Language="F#" Value="type HttpException = class&#xA;    inherit ExternalException" />
  <AssemblyInfo>
    <AssemblyName>System.Web</AssemblyName>
    <AssemblyVersion>1.0.5000.0</AssemblyVersion>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Runtime.InteropServices.ExternalException</BaseTypeName>
  </Base>
  <Interfaces />
  <Docs>
    <summary>
      <span data-ttu-id="09391-101">Describe una excepción que apareció durante el procesamiento de solicitudes HTTP.</span>
      <span class="sxs-lookup">
        <span data-stu-id="09391-101">Describes an exception that occurred during the processing of HTTP requests.</span>
      </span>
    </summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="09391-102">El <xref:System.Web.HttpException> es una clase de excepción específica de HTTP que permite a ASP.NET generar información de excepción.</span><span class="sxs-lookup"><span data-stu-id="09391-102">The <xref:System.Web.HttpException> class is an HTTP-specific exception class that enables ASP.NET to generate exception information.</span></span> <span data-ttu-id="09391-103">Para obtener más información sobre cómo iniciar y controlar las excepciones, vea [excepciones](http://msdn.microsoft.com/library/f99a1d29-a2a8-47af-9707-9909f9010735).</span><span class="sxs-lookup"><span data-stu-id="09391-103">For more information about throwing and handling exceptions, see [Exceptions](http://msdn.microsoft.com/library/f99a1d29-a2a8-47af-9707-9909f9010735).</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="09391-104">En el ejemplo de código siguiente se muestra cómo usar el <xref:System.Web.HttpException> clase para generar excepciones personalizadas.</span><span class="sxs-lookup"><span data-stu-id="09391-104">The following code example demonstrates how to use the <xref:System.Web.HttpException> class to raise customized exceptions.</span></span> <span data-ttu-id="09391-105">El `Button_Click` método determina qué botón se hizo clic en y, a continuación, llama a la `CheckNumber` o `CheckBoolean` método.</span><span class="sxs-lookup"><span data-stu-id="09391-105">The `Button_Click` method determines which button was clicked, and then calls either the `CheckNumber` or `CheckBoolean` method.</span></span> <span data-ttu-id="09391-106">Si el valor escrito por el usuario en el cuadro de texto no corresponde al tipo esperado, un <xref:System.Web.HttpException> excepción se crean y se inicia con un mensaje personalizado en función del tipo de error.</span><span class="sxs-lookup"><span data-stu-id="09391-106">If the user-entered value in the text box does not correspond to the expected type, an <xref:System.Web.HttpException> exception is created and thrown with a customized message based on the error type.</span></span>  
  
> [!IMPORTANT]
>  <span data-ttu-id="09391-107">Este ejemplo tiene un cuadro de texto que acepta datos proporcionados por el usuario, lo que puede suponer una amenaza para la seguridad.</span><span class="sxs-lookup"><span data-stu-id="09391-107">This example has a text box that accepts user input, which is a potential security threat.</span></span> <span data-ttu-id="09391-108">De forma predeterminada, ASP.NET Web Pages valida que los datos proporcionados por el usuario no incluyen elementos HTML ni de script.</span><span class="sxs-lookup"><span data-stu-id="09391-108">By default, ASP.NET Web pages validate that user input does not include script or HTML elements.</span></span> <span data-ttu-id="09391-109">Para más información, consulte [Información general sobre los ataques mediante scripts](http://msdn.microsoft.com/library/772c7312-211a-4eb3-8d6e-eec0aa1dcc07).</span><span class="sxs-lookup"><span data-stu-id="09391-109">For more information, see [Script Exploits Overview](http://msdn.microsoft.com/library/772c7312-211a-4eb3-8d6e-eec0aa1dcc07).</span></span>  
  
 [!code-aspx-csharp[HttpException#1](~/samples/snippets/csharp/VS_Snippets_WebNet/HttpException/cs/httpexceptioncs.aspx#1)]
 [!code-aspx-vb[HttpException#1](~/samples/snippets/visualbasic/VS_Snippets_WebNet/HttpException/vb/httpexceptionvb.aspx#1)]  
  
 ]]></format>
    </remarks>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Web</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary>
          <span data-ttu-id="09391-110">Inicializa una nueva instancia de la clase <see cref="T:System.Web.HttpException" />.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-110">Initializes a new instance of the <see cref="T:System.Web.HttpException" /> class.</span>
          </span>
        </summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public HttpException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.HttpException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; HttpException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="09391-111">Inicializa una nueva instancia de la clase <see cref="T:System.Web.HttpException" /> y crea un objeto <see cref="T:System.Web.HttpException" /> vacío.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-111">Initializes a new instance of the <see cref="T:System.Web.HttpException" /> class and creates an empty <see cref="T:System.Web.HttpException" /> object.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="09391-112">Cuando el control de excepciones, puede ser útil capturar una serie de excepciones relacionadas con la excepción externa producida en respuesta a una excepción interna.</span><span class="sxs-lookup"><span data-stu-id="09391-112">When handling exceptions, it can be useful to capture a series of related exceptions with the outer exception being thrown in response to an inner exception.</span></span>  
  
 <span data-ttu-id="09391-113">Una referencia a la excepción interna que produjo la excepción externa está disponible desde el <xref:System.Exception.InnerException%2A> propiedad de la excepción externa.</span><span class="sxs-lookup"><span data-stu-id="09391-113">A reference to the inner exception that caused the outer exception is available from the <xref:System.Exception.InnerException%2A> property of the outer exception.</span></span> <span data-ttu-id="09391-114">Este mecanismo conserva la información de error que se lleva a las excepciones anteriores, incluidas las excepciones originales, mientras que le permite crear excepciones externas más significativas.</span><span class="sxs-lookup"><span data-stu-id="09391-114">This mechanism preserves the error information that is carried by earlier exceptions, including the original exceptions, while allowing you to create more meaningful outer exceptions.</span></span> <span data-ttu-id="09391-115">Para obtener más información, vea <xref:System.Exception.InnerException%2A>.</span><span class="sxs-lookup"><span data-stu-id="09391-115">For more information, see <xref:System.Exception.InnerException%2A>.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="09391-116">En el ejemplo de código siguiente se muestra el <xref:System.Web.HttpException.%23ctor%2A> constructor de la <xref:System.Web.HttpException> clase.</span><span class="sxs-lookup"><span data-stu-id="09391-116">The following code example demonstrates the <xref:System.Web.HttpException.%23ctor%2A> constructor of the <xref:System.Web.HttpException> class.</span></span> <span data-ttu-id="09391-117">El `CheckNumber` método acepta un valor escrito por el usuario a través de un cuadro de texto y comprueba si es un entero.</span><span class="sxs-lookup"><span data-stu-id="09391-117">The `CheckNumber` method accepts a user-entered value through a text box and checks whether it is an integer.</span></span> <span data-ttu-id="09391-118">Si el valor no es un entero, se produce una excepción y, a continuación, un nuevo <xref:System.Web.HttpException> objeto se crea y se produce.</span><span class="sxs-lookup"><span data-stu-id="09391-118">If the value is not an integer, an exception is thrown, and then a new <xref:System.Web.HttpException> object is created and thrown.</span></span> <span data-ttu-id="09391-119">Que se detecta la excepción en el `Button_Click` controlador de eventos y el mensaje de error se muestra en el explorador.</span><span class="sxs-lookup"><span data-stu-id="09391-119">That exception is caught in the `Button_Click` event handler and the error message is displayed on the browser.</span></span>  
  
> [!IMPORTANT]
>  <span data-ttu-id="09391-120">Este ejemplo tiene un cuadro de texto que acepta datos proporcionados por el usuario, lo que puede suponer una amenaza para la seguridad.</span><span class="sxs-lookup"><span data-stu-id="09391-120">This example has a text box that accepts user input, which is a potential security threat.</span></span> <span data-ttu-id="09391-121">De forma predeterminada, ASP.NET Web Pages valida que los datos proporcionados por el usuario no incluyen elementos HTML ni de script.</span><span class="sxs-lookup"><span data-stu-id="09391-121">By default, ASP.NET Web pages validate that user input does not include script or HTML elements.</span></span> <span data-ttu-id="09391-122">Para más información, consulte [Información general sobre los ataques mediante scripts](http://msdn.microsoft.com/library/772c7312-211a-4eb3-8d6e-eec0aa1dcc07).</span><span class="sxs-lookup"><span data-stu-id="09391-122">For more information, see [Script Exploits Overview](http://msdn.microsoft.com/library/772c7312-211a-4eb3-8d6e-eec0aa1dcc07).</span></span>  
  
 [!code-aspx-csharp[HttpException_Constructor1#1](~/samples/snippets/csharp/VS_Snippets_WebNet/HttpException_Constructor1/CS/httpexception_constructor1.cs.aspx#1)]
 [!code-aspx-vb[HttpException_Constructor1#1](~/samples/snippets/visualbasic/VS_Snippets_WebNet/HttpException_Constructor1/VB/httpexception_constructor1.vb.aspx#1)]  
  
 ]]></format>
        </remarks>
        <altmember cref="P:System.Exception.InnerException" />
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public HttpException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.HttpException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; HttpException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.Web.HttpException : string -&gt; System.Web.HttpException" Usage="new System.Web.HttpException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="09391-123">Mensaje de error que se muestra al cliente cuando se produce la excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-123">The error message displayed to the client when the exception is thrown.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="09391-124">Inicializa una nueva instancia de la clase <see cref="T:System.Web.HttpException" /> utilizando el mensaje de error especificado.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-124">Initializes a new instance of the <see cref="T:System.Web.HttpException" /> class using a supplied error message.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Examples  
 <span data-ttu-id="09391-125">En el ejemplo de código siguiente se muestra el <xref:System.Web.HttpException.%23ctor%2A> constructor de la <xref:System.Web.HttpException> clase.</span><span class="sxs-lookup"><span data-stu-id="09391-125">The following code example demonstrates the <xref:System.Web.HttpException.%23ctor%2A> constructor of the <xref:System.Web.HttpException> class.</span></span> <span data-ttu-id="09391-126">Un <xref:System.Web.HttpException> se produce si un valor escrito por el usuario es 0.</span><span class="sxs-lookup"><span data-stu-id="09391-126">An <xref:System.Web.HttpException> is thrown if a user-entered value is 0.</span></span>  
  
> [!IMPORTANT]
>  <span data-ttu-id="09391-127">Este ejemplo tiene un cuadro de texto que acepta datos proporcionados por el usuario, lo que puede suponer una amenaza para la seguridad.</span><span class="sxs-lookup"><span data-stu-id="09391-127">This example has a text box that accepts user input, which is a potential security threat.</span></span> <span data-ttu-id="09391-128">De forma predeterminada, ASP.NET Web Pages valida que los datos proporcionados por el usuario no incluyen elementos HTML ni de script.</span><span class="sxs-lookup"><span data-stu-id="09391-128">By default, ASP.NET Web pages validate that user input does not include script or HTML elements.</span></span> <span data-ttu-id="09391-129">Para más información, consulte [Información general sobre los ataques mediante scripts](http://msdn.microsoft.com/library/772c7312-211a-4eb3-8d6e-eec0aa1dcc07).</span><span class="sxs-lookup"><span data-stu-id="09391-129">For more information, see [Script Exploits Overview](http://msdn.microsoft.com/library/772c7312-211a-4eb3-8d6e-eec0aa1dcc07).</span></span>  
  
 [!code-csharp[Classic HttpException.HttpException1 Example#1](~/samples/snippets/csharp/VS_Snippets_WebNet/Classic HttpException.HttpException1 Example/CS/source.cs#1)]
 [!code-vb[Classic HttpException.HttpException1 Example#1](~/samples/snippets/visualbasic/VS_Snippets_WebNet/Classic HttpException.HttpException1 Example/VB/source.vb#1)]  
  
 ]]></format>
        </remarks>
        <altmember cref="P:System.Exception.InnerException" />
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public HttpException (int httpCode, string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(int32 httpCode, string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.HttpException.#ctor(System.Int32,System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (httpCode As Integer, message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; HttpException(int httpCode, System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.Web.HttpException : int * string -&gt; System.Web.HttpException" Usage="new System.Web.HttpException (httpCode, message)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="httpCode" Type="System.Int32" />
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="httpCode">
          <span data-ttu-id="09391-130">Código de estado de respuesta HTTP enviado al cliente que corresponde a este error.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-130">The HTTP response status code sent to the client corresponding to this error.</span>
          </span>
        </param>
        <param name="message">
          <span data-ttu-id="09391-131">Mensaje de error que se muestra al cliente cuando se produce la excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-131">The error message displayed to the client when the exception is thrown.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="09391-132">Inicializa una nueva instancia de la clase <see cref="T:System.Web.HttpException" /> utilizando un código de estado de respuesta HTTP y un mensaje de error.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-132">Initializes a new instance of the <see cref="T:System.Web.HttpException" /> class using an HTTP response status code and an error message.</span>
          </span>
        </summary>
        <remarks>To be added.</remarks>
        <altmember cref="P:System.Exception.InnerException" />
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected HttpException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.HttpException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; HttpException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new System.Web.HttpException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; System.Web.HttpException" Usage="new System.Web.HttpException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info">
          <span data-ttu-id="09391-133">
            <see cref="T:System.Runtime.Serialization.SerializationInfo" /> que contiene los datos serializados del objeto que hacen referencia a la excepción que se va a producir.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-133">The <see cref="T:System.Runtime.Serialization.SerializationInfo" /> that holds the serialized object data about the exception being thrown.</span>
          </span>
        </param>
        <param name="context">
          <span data-ttu-id="09391-134">
            <see cref="T:System.Runtime.Serialization.StreamingContext" /> que contiene la información contextual sobre el origen o el destino.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-134">The <see cref="T:System.Runtime.Serialization.StreamingContext" /> that holds the contextual information about the source or destination.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="09391-135">Inicializa una nueva instancia de la clase <see cref="T:System.Web.HttpException" /> con datos serializados.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-135">Initializes a new instance of the <see cref="T:System.Web.HttpException" /> class with serialized data.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="09391-136">El <xref:System.Web.HttpException.%23ctor%2A> constructor se llama durante la deserialización para reconstituir el objeto de excepción que se transmite a través de una secuencia.</span><span class="sxs-lookup"><span data-stu-id="09391-136">The <xref:System.Web.HttpException.%23ctor%2A> constructor is called during deserialization to reconstitute the exception object that is transmitted over a stream.</span></span> <span data-ttu-id="09391-137">Para obtener más información, consulte [serialización SOAP y XML](~/docs/standard/serialization/xml-and-soap-serialization.md).</span><span class="sxs-lookup"><span data-stu-id="09391-137">For more information, see [XML and SOAP Serialization](~/docs/standard/serialization/xml-and-soap-serialization.md).</span></span>  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Runtime.Serialization.SerializationInfo" />
        <altmember cref="T:System.Runtime.Serialization.StreamingContext" />
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public HttpException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.HttpException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; HttpException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new System.Web.HttpException : string * Exception -&gt; System.Web.HttpException" Usage="new System.Web.HttpException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="09391-138">Mensaje de error que se muestra al cliente cuando se produce la excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-138">The error message displayed to the client when the exception is thrown.</span>
          </span>
        </param>
        <param name="innerException">
          <span data-ttu-id="09391-139">Propiedad <see cref="P:System.Exception.InnerException" />, si hay alguna, que ha producido la excepción actual.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-139">The <see cref="P:System.Exception.InnerException" />, if any, that threw the current exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="09391-140">Inicializa una nueva instancia de la clase <see cref="T:System.Web.HttpException" /> utilizando un mensaje de error y la propiedad <see cref="P:System.Exception.InnerException" />.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-140">Initializes a new instance of the <see cref="T:System.Web.HttpException" /> class using an error message and the <see cref="P:System.Exception.InnerException" /> property.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="09391-141">Cuando el control de excepciones, puede ser útil capturar una serie de excepciones relacionadas con la excepción externa producida en respuesta a una excepción interna.</span><span class="sxs-lookup"><span data-stu-id="09391-141">When handling exceptions, it can be useful to capture a series of related exceptions with the outer exception being thrown in response to an inner exception.</span></span>  
  
 <span data-ttu-id="09391-142">Una referencia a la excepción interna que produjo la excepción externa está disponible desde el <xref:System.Exception.InnerException%2A> propiedad de la excepción externa.</span><span class="sxs-lookup"><span data-stu-id="09391-142">A reference to the inner exception that caused the outer exception is available from the <xref:System.Exception.InnerException%2A> property of the outer exception.</span></span> <span data-ttu-id="09391-143">Este mecanismo conserva la información de error que se lleva a las excepciones anteriores, incluidas las excepciones originales, mientras que le permite crear excepciones externas más significativas.</span><span class="sxs-lookup"><span data-stu-id="09391-143">This mechanism preserves the error information that is carried by earlier exceptions, including the original exceptions, while allowing you to create more meaningful outer exceptions.</span></span> <span data-ttu-id="09391-144">Para obtener más información, vea <xref:System.Exception.InnerException%2A>.</span><span class="sxs-lookup"><span data-stu-id="09391-144">For more information, see <xref:System.Exception.InnerException%2A>.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="09391-145">En el ejemplo de código siguiente se muestra el <xref:System.Web.HttpException.%23ctor%2A> constructor de la <xref:System.Web.HttpException> clase.</span><span class="sxs-lookup"><span data-stu-id="09391-145">The following code example demonstrates the <xref:System.Web.HttpException.%23ctor%2A> constructor of the <xref:System.Web.HttpException> class.</span></span> <span data-ttu-id="09391-146">El `CheckNumber` método acepta un valor escrito por el usuario a través de un cuadro de texto y comprueba si es un entero.</span><span class="sxs-lookup"><span data-stu-id="09391-146">The `CheckNumber` method accepts a user-entered value through a text box and checks whether it is an integer.</span></span> <span data-ttu-id="09391-147">Si el valor no es un entero, se produce una excepción y, a continuación, en el bloque catch, un nuevo <xref:System.Web.HttpException> objeto se crea y se produce.</span><span class="sxs-lookup"><span data-stu-id="09391-147">If the value is not an integer, an exception is thrown, and then in the catch block, a new <xref:System.Web.HttpException> object is created and thrown.</span></span> <span data-ttu-id="09391-148">Que se detecta la excepción en el `Button_Click` controlador de eventos y el mensaje de error se muestra en el explorador.</span><span class="sxs-lookup"><span data-stu-id="09391-148">That exception is caught in the `Button_Click` event handler and the error message is displayed on the browser.</span></span>  
  
> [!IMPORTANT]
>  <span data-ttu-id="09391-149">Este ejemplo tiene un cuadro de texto que acepta datos proporcionados por el usuario, lo que puede suponer una amenaza para la seguridad.</span><span class="sxs-lookup"><span data-stu-id="09391-149">This example has a text box that accepts user input, which is a potential security threat.</span></span> <span data-ttu-id="09391-150">De forma predeterminada, ASP.NET Web Pages valida que los datos proporcionados por el usuario no incluyen elementos HTML ni de script.</span><span class="sxs-lookup"><span data-stu-id="09391-150">By default, ASP.NET Web pages validate that user input does not include script or HTML elements.</span></span> <span data-ttu-id="09391-151">Para más información, consulte [Información general sobre los ataques mediante scripts](http://msdn.microsoft.com/library/772c7312-211a-4eb3-8d6e-eec0aa1dcc07).</span><span class="sxs-lookup"><span data-stu-id="09391-151">For more information, see [Script Exploits Overview](http://msdn.microsoft.com/library/772c7312-211a-4eb3-8d6e-eec0aa1dcc07).</span></span>  
  
 [!code-aspx-csharp[HttpException_Constructor2#1](~/samples/snippets/csharp/VS_Snippets_WebNet/HttpException_Constructor2/CS/httpexception_constructor2.cs.aspx#1)]
 [!code-aspx-vb[HttpException_Constructor2#1](~/samples/snippets/visualbasic/VS_Snippets_WebNet/HttpException_Constructor2/VB/httpexception_constructor2.vb.aspx#1)]  
  
 ]]></format>
        </remarks>
        <altmember cref="P:System.Exception.InnerException" />
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public HttpException (string message, int hr);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, int32 hr) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.HttpException.#ctor(System.String,System.Int32)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, hr As Integer)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; HttpException(System::String ^ message, int hr);" />
      <MemberSignature Language="F#" Value="new System.Web.HttpException : string * int -&gt; System.Web.HttpException" Usage="new System.Web.HttpException (message, hr)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="hr" Type="System.Int32" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="09391-152">Mensaje de error que se muestra al cliente cuando se produce la excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-152">The error message displayed to the client when the exception is thrown.</span>
          </span>
        </param>
        <param name="hr">
          <span data-ttu-id="09391-153">Código de excepción que define el error.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-153">The exception code that defines the error.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="09391-154">Inicializa una nueva instancia de la clase <see cref="T:System.Web.HttpException" /> utilizando un mensaje de error y un código de excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-154">Initializes a new instance of the <see cref="T:System.Web.HttpException" /> class using an error message and an exception code.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Examples  
 <span data-ttu-id="09391-155">En el ejemplo de código siguiente se muestra el <xref:System.Web.HttpException.%23ctor%2A> constructor de la <xref:System.Web.HttpException> clase.</span><span class="sxs-lookup"><span data-stu-id="09391-155">The following code example demonstrates the <xref:System.Web.HttpException.%23ctor%2A> constructor of the <xref:System.Web.HttpException> class.</span></span> <span data-ttu-id="09391-156">Un <xref:System.Web.HttpException> excepción se produce si un valor escrito por el usuario es 0.</span><span class="sxs-lookup"><span data-stu-id="09391-156">An <xref:System.Web.HttpException> exception is thrown if a user-entered value is 0.</span></span>  
  
> [!IMPORTANT]
>  <span data-ttu-id="09391-157">Este ejemplo tiene un cuadro de texto que acepta datos proporcionados por el usuario, lo que puede suponer una amenaza para la seguridad.</span><span class="sxs-lookup"><span data-stu-id="09391-157">This example has a text box that accepts user input, which is a potential security threat.</span></span> <span data-ttu-id="09391-158">De forma predeterminada, ASP.NET Web Pages valida que los datos proporcionados por el usuario no incluyen elementos HTML ni de script.</span><span class="sxs-lookup"><span data-stu-id="09391-158">By default, ASP.NET Web pages validate that user input does not include script or HTML elements.</span></span> <span data-ttu-id="09391-159">Para más información, consulte [Información general sobre los ataques mediante scripts](http://msdn.microsoft.com/library/772c7312-211a-4eb3-8d6e-eec0aa1dcc07).</span><span class="sxs-lookup"><span data-stu-id="09391-159">For more information, see [Script Exploits Overview](http://msdn.microsoft.com/library/772c7312-211a-4eb3-8d6e-eec0aa1dcc07).</span></span>  
  
 [!code-csharp[Classic HttpException.HttpException2 Example#1](~/samples/snippets/csharp/VS_Snippets_WebNet/Classic HttpException.HttpException2 Example/CS/source.cs#1)]
 [!code-vb[Classic HttpException.HttpException2 Example#1](~/samples/snippets/visualbasic/VS_Snippets_WebNet/Classic HttpException.HttpException2 Example/VB/source.vb#1)]  
  
 ]]></format>
        </remarks>
        <altmember cref="P:System.Exception.InnerException" />
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public HttpException (int httpCode, string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(int32 httpCode, string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.HttpException.#ctor(System.Int32,System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (httpCode As Integer, message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; HttpException(int httpCode, System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new System.Web.HttpException : int * string * Exception -&gt; System.Web.HttpException" Usage="new System.Web.HttpException (httpCode, message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="httpCode" Type="System.Int32" />
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="httpCode">
          <span data-ttu-id="09391-160">Código de estado de respuesta HTTP que se muestra en el cliente.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-160">The HTTP response status code displayed on the client.</span>
          </span>
        </param>
        <param name="message">
          <span data-ttu-id="09391-161">Mensaje de error que se muestra al cliente cuando se produce la excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-161">The error message displayed to the client when the exception is thrown.</span>
          </span>
        </param>
        <param name="innerException">
          <span data-ttu-id="09391-162">Propiedad <see cref="P:System.Exception.InnerException" />, si hay alguna, que ha producido la excepción actual.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-162">The <see cref="P:System.Exception.InnerException" />, if any, that threw the current exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="09391-163">Inicializa una nueva instancia de la clase <see cref="T:System.Web.HttpException" /> utilizando un código de estado de respuesta HTTP, un mensaje de error y la propiedad <see cref="P:System.Exception.InnerException" />.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-163">Initializes a new instance of the <see cref="T:System.Web.HttpException" /> class using an HTTP response status code, an error message, and the <see cref="P:System.Exception.InnerException" /> property.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="09391-164">Cuando el control de excepciones, puede ser útil capturar una serie de excepciones relacionadas con la excepción externa producida en respuesta a una excepción interna.</span><span class="sxs-lookup"><span data-stu-id="09391-164">When handling exceptions, it can be useful to capture a series of related exceptions with the outer exception being thrown in response to an inner exception.</span></span>  
  
 <span data-ttu-id="09391-165">Una referencia a la excepción interna que produjo la excepción externa está disponible desde el <xref:System.Exception.InnerException%2A> propiedad de la excepción externa.</span><span class="sxs-lookup"><span data-stu-id="09391-165">A reference to the inner exception that caused the outer exception is available from the <xref:System.Exception.InnerException%2A> property of the outer exception.</span></span> <span data-ttu-id="09391-166">Este mecanismo conserva la información de error que se lleva a las excepciones anteriores, incluidas las excepciones originales, mientras que le permite crear excepciones externas más significativas.</span><span class="sxs-lookup"><span data-stu-id="09391-166">This mechanism preserves the error information that is carried by earlier exceptions, including the original exceptions, while allowing you to create more meaningful outer exceptions.</span></span> <span data-ttu-id="09391-167">Para obtener más información, vea <xref:System.Exception.InnerException%2A>.</span><span class="sxs-lookup"><span data-stu-id="09391-167">For more information, see <xref:System.Exception.InnerException%2A>.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="09391-168">En el ejemplo de código siguiente se muestra el <xref:System.Web.HttpException.%23ctor%2A> constructor de la <xref:System.Web.HttpException> clase.</span><span class="sxs-lookup"><span data-stu-id="09391-168">The following code example demonstrates the <xref:System.Web.HttpException.%23ctor%2A> constructor of the <xref:System.Web.HttpException> class.</span></span> <span data-ttu-id="09391-169">El `CheckNumber` método acepta un valor escrito por el usuario y comprueba si es un entero.</span><span class="sxs-lookup"><span data-stu-id="09391-169">The `CheckNumber` method accepts a user-entered value and checks whether it is an integer.</span></span> <span data-ttu-id="09391-170">Si el valor no es un entero, se produce una excepción y, a continuación, un nuevo <xref:System.Web.HttpException> se crea el objeto que contiene el código de estado de respuesta HTTP, el mensaje de excepción y cualquier excepción interna.</span><span class="sxs-lookup"><span data-stu-id="09391-170">If the value is not an integer, an exception is thrown, and then a new <xref:System.Web.HttpException> object containing the HTTP response status code, the exception's message, and any inner exception is created.</span></span> <span data-ttu-id="09391-171">Que se detecta la excepción en el `Button_Click` se muestran el controlador de eventos y el mensaje de error, el código de error y la excepción interna.</span><span class="sxs-lookup"><span data-stu-id="09391-171">That exception is caught in the `Button_Click` event handler and the error message, error code, and inner exception are displayed.</span></span>  
  
> [!IMPORTANT]
>  <span data-ttu-id="09391-172">Este ejemplo tiene un cuadro de texto que acepta datos proporcionados por el usuario, lo que puede suponer una amenaza para la seguridad.</span><span class="sxs-lookup"><span data-stu-id="09391-172">This example has a text box that accepts user input, which is a potential security threat.</span></span> <span data-ttu-id="09391-173">De forma predeterminada, ASP.NET Web Pages valida que los datos proporcionados por el usuario no incluyen elementos HTML ni de script.</span><span class="sxs-lookup"><span data-stu-id="09391-173">By default, ASP.NET Web pages validate that user input does not include script or HTML elements.</span></span> <span data-ttu-id="09391-174">Para más información, consulte [Información general sobre los ataques mediante scripts](http://msdn.microsoft.com/library/772c7312-211a-4eb3-8d6e-eec0aa1dcc07).</span><span class="sxs-lookup"><span data-stu-id="09391-174">For more information, see [Script Exploits Overview](http://msdn.microsoft.com/library/772c7312-211a-4eb3-8d6e-eec0aa1dcc07).</span></span>  
  
 [!code-aspx-csharp[HttpException_Constructor3#1](~/samples/snippets/csharp/VS_Snippets_WebNet/HttpException_Constructor3/CS/httpexception_constructor3.cs.aspx#1)]
 [!code-aspx-vb[HttpException_Constructor3#1](~/samples/snippets/visualbasic/VS_Snippets_WebNet/HttpException_Constructor3/VB/httpexception_constructor3.vb.aspx#1)]  
  
 ]]></format>
        </remarks>
        <altmember cref="P:System.Exception.InnerException" />
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public HttpException (int httpCode, string message, int hr);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(int32 httpCode, string message, int32 hr) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.HttpException.#ctor(System.Int32,System.String,System.Int32)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (httpCode As Integer, message As String, hr As Integer)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; HttpException(int httpCode, System::String ^ message, int hr);" />
      <MemberSignature Language="F#" Value="new System.Web.HttpException : int * string * int -&gt; System.Web.HttpException" Usage="new System.Web.HttpException (httpCode, message, hr)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="httpCode" Type="System.Int32" />
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="hr" Type="System.Int32" />
      </Parameters>
      <Docs>
        <param name="httpCode">
          <span data-ttu-id="09391-175">Código de estado de respuesta HTTP que se muestra en el cliente.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-175">The HTTP response status code displayed on the client.</span>
          </span>
        </param>
        <param name="message">
          <span data-ttu-id="09391-176">Mensaje de error que se muestra al cliente cuando se produce la excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-176">The error message displayed to the client when the exception is thrown.</span>
          </span>
        </param>
        <param name="hr">
          <span data-ttu-id="09391-177">Código de excepción que define el error.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-177">The exception code that defines the error.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="09391-178">Inicializa una nueva instancia de la clase <see cref="T:System.Web.HttpException" /> utilizando un código de estado de respuesta HTTP, un mensaje de error y un código de excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-178">Initializes a new instance of the <see cref="T:System.Web.HttpException" /> class using an HTTP response status code, an error message, and an exception code.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Examples  
 <span data-ttu-id="09391-179">En el ejemplo de código siguiente se muestra el <xref:System.Web.HttpException.%23ctor%2A> constructor de la <xref:System.Web.HttpException> clase.</span><span class="sxs-lookup"><span data-stu-id="09391-179">The following code example demonstrates the <xref:System.Web.HttpException.%23ctor%2A> constructor of the <xref:System.Web.HttpException> class.</span></span> <span data-ttu-id="09391-180">La información de nombre y correo electrónico del usuario se escriben por el usuario en los cuadros de texto proporcionado.</span><span class="sxs-lookup"><span data-stu-id="09391-180">The user name and email information are entered by the user in the provided text boxes.</span></span> <span data-ttu-id="09391-181">Si cualquiera de los cuadros de texto se deja en blanco, un <xref:System.Web.HttpException> objeto se crea y se produce.</span><span class="sxs-lookup"><span data-stu-id="09391-181">If any of the text boxes are left blank, an <xref:System.Web.HttpException> object is created and thrown.</span></span> <span data-ttu-id="09391-182">El código de error de la <xref:System.Web.HttpException> se obtiene mediante la <xref:System.Web.HttpException.GetHttpCode%2A> método y se muestra en la página Web.</span><span class="sxs-lookup"><span data-stu-id="09391-182">The error code of the <xref:System.Web.HttpException> is obtained by the <xref:System.Web.HttpException.GetHttpCode%2A> method and displayed on the Web page.</span></span>  
  
> [!IMPORTANT]
>  <span data-ttu-id="09391-183">Este ejemplo tiene un cuadro de texto que acepta datos proporcionados por el usuario, lo que puede suponer una amenaza para la seguridad.</span><span class="sxs-lookup"><span data-stu-id="09391-183">This example has a text box that accepts user input, which is a potential security threat.</span></span> <span data-ttu-id="09391-184">De forma predeterminada, ASP.NET Web Pages valida que los datos proporcionados por el usuario no incluyen elementos HTML ni de script.</span><span class="sxs-lookup"><span data-stu-id="09391-184">By default, ASP.NET Web pages validate that user input does not include script or HTML elements.</span></span> <span data-ttu-id="09391-185">Para más información, consulte [Información general sobre los ataques mediante scripts](http://msdn.microsoft.com/library/772c7312-211a-4eb3-8d6e-eec0aa1dcc07).</span><span class="sxs-lookup"><span data-stu-id="09391-185">For more information, see [Script Exploits Overview](http://msdn.microsoft.com/library/772c7312-211a-4eb3-8d6e-eec0aa1dcc07).</span></span>  
  
 [!code-aspx-csharp[HttpException_Constructor#1](~/samples/snippets/csharp/VS_Snippets_WebNet/HttpException_Constructor/CS/httpexception_constructor5.cs.aspx#1)]
 [!code-aspx-vb[HttpException_Constructor#1](~/samples/snippets/visualbasic/VS_Snippets_WebNet/HttpException_Constructor/VB/httpexception_constructor5.vb.aspx#1)]  
  
 ]]></format>
        </remarks>
        <altmember cref="P:System.Exception.InnerException" />
      </Docs>
    </Member>
    <Member MemberName="CreateFromLastError">
      <MemberSignature Language="C#" Value="public static System.Web.HttpException CreateFromLastError (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public static hidebysig class System.Web.HttpException CreateFromLastError(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.HttpException.CreateFromLastError(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Shared Function CreateFromLastError (message As String) As HttpException" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; static System::Web::HttpException ^ CreateFromLastError(System::String ^ message);" />
      <MemberSignature Language="F#" Value="static member CreateFromLastError : string -&gt; System.Web.HttpException" Usage="System.Web.HttpException.CreateFromLastError message" />
      <MemberType>Method</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Web.HttpException</ReturnType>
      </ReturnValue>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="09391-186">Mensaje de error que se muestra al cliente cuando se produce la excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-186">The error message displayed to the client when the exception is thrown.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="09391-187">Crea una nueva excepción <see cref="T:System.Web.HttpException" /> basada en el código de error devuelto por el método <see langword="GetLastError()" /> de la API Win32.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-187">Creates a new <see cref="T:System.Web.HttpException" /> exception based on the error code that is returned from the Win32 API <see langword="GetLastError()" /> method.</span>
          </span>
        </summary>
        <returns>
          <span data-ttu-id="09391-188">
            <see cref="T:System.Web.HttpException" /> basada en el código de error que se devuelve a partir de una llamada al método <see langword="GetLastError()" /> de la API Win32.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-188">An <see cref="T:System.Web.HttpException" /> based on the error code that is returned from a call to the Win32 API <see langword="GetLastError()" /> method.</span>
          </span>
        </returns>
        <remarks>To be added.</remarks>
        <altmember cref="P:System.Exception.InnerException" />
      </Docs>
    </Member>
    <Member MemberName="GetHtmlErrorMessage">
      <MemberSignature Language="C#" Value="public string GetHtmlErrorMessage ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig instance string GetHtmlErrorMessage() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.HttpException.GetHtmlErrorMessage" />
      <MemberSignature Language="VB.NET" Value="Public Function GetHtmlErrorMessage () As String" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; System::String ^ GetHtmlErrorMessage();" />
      <MemberSignature Language="F#" Value="member this.GetHtmlErrorMessage : unit -&gt; string" Usage="httpException.GetHtmlErrorMessage " />
      <MemberType>Method</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.String</ReturnType>
      </ReturnValue>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="09391-189">Obtiene el mensaje de error HTML que se va a devolver al cliente.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-189">Gets the HTML error message to return to the client.</span>
          </span>
        </summary>
        <returns>
          <span data-ttu-id="09391-190">Mensaje de error HTML.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-190">The HTML error message.</span>
          </span>
        </returns>
        <remarks>To be added.</remarks>
        <altmember cref="P:System.Exception.InnerException" />
      </Docs>
    </Member>
    <Member MemberName="GetHttpCode">
      <MemberSignature Language="C#" Value="public int GetHttpCode ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig instance int32 GetHttpCode() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.HttpException.GetHttpCode" />
      <MemberSignature Language="VB.NET" Value="Public Function GetHttpCode () As Integer" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; int GetHttpCode();" />
      <MemberSignature Language="F#" Value="member this.GetHttpCode : unit -&gt; int" Usage="httpException.GetHttpCode " />
      <MemberType>Method</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <ReturnValue>
        <ReturnType>System.Int32</ReturnType>
      </ReturnValue>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="09391-191">Obtiene el código de estado de respuesta HTTP que se va a devolver al cliente.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-191">Gets the HTTP response status code to return to the client.</span>
          </span>
        </summary>
        <returns>
          <span data-ttu-id="09391-192">Es el código HTTP distinto de cero que representa la excepción o el código <see cref="P:System.Exception.InnerException" />; de lo contrario, es el código de estado de respuesta HTTP 500.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-192">A non-zero HTTP code representing the exception or the <see cref="P:System.Exception.InnerException" /> code; otherwise, HTTP response status code 500.</span>
          </span>
        </returns>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="09391-193">Si hay un código HTTP distinto de cero, se devuelve.</span><span class="sxs-lookup"><span data-stu-id="09391-193">If there is a non-zero HTTP code, it is returned.</span></span> <span data-ttu-id="09391-194">En caso contrario, el <xref:System.Exception.InnerException%2A> se devuelve el código.</span><span class="sxs-lookup"><span data-stu-id="09391-194">Otherwise, the <xref:System.Exception.InnerException%2A> code is returned.</span></span> <span data-ttu-id="09391-195">Si no un <xref:System.Exception.InnerException%2A> código ni un código HTTP distinto de cero está disponible, se devuelve el código de estado de respuesta HTTP 500.</span><span class="sxs-lookup"><span data-stu-id="09391-195">If neither an <xref:System.Exception.InnerException%2A> code nor a non-zero HTTP code is available, the HTTP response status code 500 is returned.</span></span> <span data-ttu-id="09391-196">Para obtener más información sobre los códigos de estado de respuesta HTTP, vea las definiciones de código de estado de Hypertext Transfer Protocol - HTTP/1.1 en el mundo [sitio Web de Wide Web Consortium (W3C)](http://go.microsoft.com/fwlink/?linkid=37125).</span><span class="sxs-lookup"><span data-stu-id="09391-196">For more information on HTTP response status codes, see the Status Code Definitions of the Hypertext Transfer Protocol - HTTP/1.1 on the World [Wide Web Consortium (W3C) Web site](http://go.microsoft.com/fwlink/?linkid=37125).</span></span>  
  
 ]]></format>
        </remarks>
        <altmember cref="P:System.Exception.InnerException" />
      </Docs>
    </Member>
    <Member MemberName="GetObjectData">
      <MemberSignature Language="C#" Value="public override void GetObjectData (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig virtual instance void GetObjectData(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.HttpException.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Public Overrides Sub GetObjectData (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; override void GetObjectData(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="override this.GetObjectData : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; unit" Usage="httpException.GetObjectData (info, context)" />
      <MemberType>Method</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Void</ReturnType>
      </ReturnValue>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info">
          <span data-ttu-id="09391-197">
            <see cref="T:System.Runtime.Serialization.SerializationInfo" /> que contiene los datos serializados del objeto que hacen referencia a la excepción que se va a producir.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-197">The <see cref="T:System.Runtime.Serialization.SerializationInfo" /> that holds the serialized object data about the exception being thrown.</span>
          </span>
        </param>
        <param name="context">
          <span data-ttu-id="09391-198">
            <see cref="T:System.Runtime.Serialization.StreamingContext" /> que contiene la información contextual sobre el origen o el destino.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-198">The <see cref="T:System.Runtime.Serialization.StreamingContext" /> that holds the contextual information about the source or destination.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="09391-199">Obtiene información sobre la excepción y la agrega al objeto <see cref="T:System.Runtime.Serialization.SerializationInfo" />.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-199">Gets information about the exception and adds it to the <see cref="T:System.Runtime.Serialization.SerializationInfo" /> object.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="09391-200">El <xref:System.Web.HttpException.GetObjectData%2A> método agrega información sobre la excepción a la <xref:System.Runtime.Serialization.SerializationInfo> objeto destinado para la serialización.</span><span class="sxs-lookup"><span data-stu-id="09391-200">The <xref:System.Web.HttpException.GetObjectData%2A> method adds information about the exception to the <xref:System.Runtime.Serialization.SerializationInfo> object that is targeted for serialization.</span></span> <span data-ttu-id="09391-201">Durante la deserialización, la excepción se reconstituye a partir del <xref:System.Runtime.Serialization.SerializationInfo> que se transmite a través de la secuencia.</span><span class="sxs-lookup"><span data-stu-id="09391-201">During deserialization, the exception is reconstituted from the <xref:System.Runtime.Serialization.SerializationInfo> that is transmitted over the stream.</span></span>  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Runtime.Serialization.SerializationInfo" />
        <altmember cref="T:System.Runtime.Serialization.StreamingContext" />
      </Docs>
    </Member>
    <Member MemberName="WebEventCode">
      <MemberSignature Language="C#" Value="public int WebEventCode { get; }" />
      <MemberSignature Language="ILAsm" Value=".property instance int32 WebEventCode" />
      <MemberSignature Language="DocId" Value="P:System.Web.HttpException.WebEventCode" />
      <MemberSignature Language="VB.NET" Value="Public ReadOnly Property WebEventCode As Integer" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; property int WebEventCode { int get(); };" />
      <MemberSignature Language="F#" Value="member this.WebEventCode : int" Usage="System.Web.HttpException.WebEventCode" />
      <MemberType>Property</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>get: System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>set: System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <ReturnValue>
        <ReturnType>System.Int32</ReturnType>
      </ReturnValue>
      <Docs>
        <summary>
          <span data-ttu-id="09391-202">Obtiene los códigos de evento asociados a la excepción HTTP.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-202">Gets the event codes that are associated with the HTTP exception.</span>
          </span>
        </summary>
        <value>
          <span data-ttu-id="09391-203">Entero que representa un código de evento web.</span>
          <span class="sxs-lookup">
            <span data-stu-id="09391-203">An integer representing a Web event code.</span>
          </span>
        </value>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
  </Members>
</Type>